src/app/qgsrelationaddpolymorphicdialog.cpp

```diff
@@ -239,7 +236,7 @@ bool QgsRelationAddPolymorphicDialog::isDefinitionValid()
 
   for ( int i = 0, l = mFieldsMappingTable->rowCount(); i < l; i++ )
   {
-    isValid &= !static_cast<QLineEdit *>( mFieldsMappingTable->cellWidget( i, 0 ) )->text().isNull();
+    isValid &= static_cast<QComboBox *>( mFieldsMappingTable->cellWidget( i, 0 ) )->currentData().toInt() != -1;
```

Comment:
```suggestion
    isValid &= qgis::down_cast<QComboBox *>( mFieldsMappingTable->cellWidget( i, 0 ) )->currentData().toInt() != -1;
```
Same for below line

Reply:
Thank you @rouault , i was not aware of this! I have applied the changes in last commit